plugins/woocommerce/src/StoreApi/Authentication.php

```diff
@@ -69,7 +85,7 @@ public function send_cors_headers( $value, $result, $request ) {
 
 		// Allow preflight requests, certain http origins, and any origin if a cart token is present. Preflight requests
 		// are allowed because we'll be unable to validate cart token headers at that point.
-		if ( $this->is_preflight() || $this->has_valid_cart_token( $request ) || is_allowed_http_origin( $origin ) ) {
+		if ( $this->is_preflight() || $this->has_valid_cart_token( $request->get_header( 'Cart-Token' ) ) || is_allowed_http_origin( $origin ) ) {
```

Comment:
Not sure why I added the redundant has_valid_cart_token method, but it probably can get deleted and replaced with validate. Also better than making it public, the last thing we'd want would be third parties using a redundant method, that would be a pet to maintain forever.

```suggestion
		if ( $this->is_preflight() || JsonWebToken::validate( $this->get_cart_token(), $this->get_cart_token_secret() ) || is_allowed_http_origin( $origin ) ) {
```

These conditions can also be reordered by performance.

Reply:
https://github.com/woocommerce/woocommerce/pull/55047/commits/84cf630021cd8724c4b437e29243433d8d95f364